ext/erb/erb.c

```diff
@@ -38,6 +38,12 @@ escaped_length(VALUE str)
 static VALUE
 optimized_escape_html(VALUE str)
 {
+    // Optimize the most common, no-escape case with strpbrk(3). Not using it after
+    // this because calling a C function many times could be slower for some cases.
+    if (strpbrk(RSTRING_PTR(str), "'&\"<>") == NULL) {
```

Comment:
I get not calling `strpbrk` after that, but if it returned a pointer on the first call we could use that to directly skip to the first char to escape, no?

Maybe not worth the extra complexity?

Reply:
I tried writing it for a bit and then stopped doing it for the extra complexity. If someone brings a simple enough patch and it's meaningfully better in benchmarks, I'd merge it though. Also note that there's https://github.com/ruby/erb/pull/30/commits/f58476b9b74d91b08871775a132a350ae8840f77 too now.

Reply:
Interesting. Are your benchmarks committed anywhere? I may try that if I get bored in the plane.

Reply:
Please use https://github.com/ruby/ruby/commit/f276d5a7fe28ae4fd2934af4befd920bb78cfa9e for it.